Source/Waterfall/Modules/ModuleWaterfallFX.cs

```diff
@@ -16,153 +18,127 @@ public class ModuleWaterfallFX : PartModule
     [KSPField(isPersistant = false)] public bool useRelativeScaling;
 
 
-    protected Dictionary<string, WaterfallController> allControllers;
+    protected readonly Dictionary<string, WaterfallController> allControllers = new(16);
+    protected readonly List<WaterfallEffect> allFX = new(16);
+    protected readonly List<WaterfallEffectTemplate> allTemplates = new(16);
+    protected readonly List<Renderer> allRenderers = new(128);
 
-    protected List<WaterfallEffect>         allFX;
-    protected List<WaterfallEffectTemplate> allTemplates;
-
-    protected bool initialized;
+    protected bool started;
     private   bool isHDR;
+    protected ConfigNode config;
+    protected ConfigNode[] effectsNodes = { };
+    protected ConfigNode[] templatesNodes = { };
 
     public List<WaterfallEffect> FX => allFX;
 
     public List<WaterfallEffectTemplate> Templates => allTemplates;
 
     public List<WaterfallController> Controllers => allControllers.Values.ToList();
+    public Dictionary<string, WaterfallController> AllControllersDict => allControllers;
+
+    public override void OnAwake()
+    {
+      base.OnAwake();
+      if (HighLogic.LoadedScene == GameScenes.LOADING) Settings.Load();
```

Comment:
Why are global load settings triggered from a module?

Reply:
Good question.
I get this early in my log with 0.8.1 release I believe because the current method is trying to access the GameDatabase before MM has actually loaded it.  So I deferred getting the settings until the first thing I thought could touch them, which is the PM's OnAwake().  I'm not sure of a better pattern that will load these before the PartCompiler starts running.  Writing a Custom Loader and inserting just before the PartCompiler might be correct but seems overkill.
```
[ERR 10:45:13.205] Cannot find config in file : WATERFALL_SETTINGS
	UnityEngine.DebugLogHandler:LogFormat(LogType, Object, String, Object[])
	ModuleManager.UnityLogHandle.InterceptLogHandler:LogFormat(LogType, Object, String, Object[])
	UnityEngine.Debug:LogError(Object)
	UrlFile:GetConfig(String)
	UrlFile:ContainsConfig(String)
	UrlDir:GetConfig(UrlIdentifier, Int32)
	UrlDir:GetConfig(UrlIdentifier, Int32)
	UrlDir:GetConfig(UrlIdentifier, Int32)
	UrlDir:GetConfig(String)
	GameDatabase:GetConfigNode(String)
	Waterfall.Settings:Load()
	Waterfall.Waterfall:Start()
```

Reply:
>  I'm not sure of a better pattern that will load these before the PartCompiler starts running.
This can be done with a `ModuleManagerPostLoad()` callback

Reply:
Implemented in https://github.com/post-kerbin-mining-corporation/Waterfall/pull/104/commits/b4f425acdfcc1402f2045d2d13481a8518d28b9a